src/analyze_fs.c

```diff
@@ -387,14 +387,17 @@ struct manifest *full_manifest_from_directory(int version)
 {
 	struct manifest *manifest;
 	char *dir;
+	int numthreads = getenv("SWUPD_NUM_THREADS") ?
+		atoi(getenv("SWUPD_NUM_THREADS")) :
+		sysconf(_SC_NPROCESSORS_ONLN);
```

Comment:
Also since this code is duplicated in three places it would seem reasonable to throw it in a macro/function I think.